src/hints/execution.rs

```diff
@@ -229,6 +228,46 @@ pub const GET_CONTRACT_ADDRESS_STATE_ENTRY: &str = indoc! {r#"
     # of this function.
     ids.state_entry = __dict_manager.get_dict(ids.contract_state_changes)[ids.contract_address]"#
 };
+
+pub fn get_contract_address_state_entry(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let dict_ptr = get_ptr_from_var_name(vars::ids::CONTRACT_STATE_CHANGES, vm, ids_data, ap_tracking)?;
+    let key = get_integer_from_var_name(vars::ids::CONTRACT_ADDRESS, vm, ids_data, ap_tracking)?;
+
+    get_state_entry(dict_ptr, key, vm, exec_scopes, ids_data, ap_tracking)?;
+
+    Ok(())
+}
+
+pub const SET_STATE_ENTRY_TO_ACCOUNT_CONTRACT_ADDRESS: &str = indoc! {r#"
+    # Fetch a state_entry in this hint and validate it in the update that comes next.
+    ids.state_entry = __dict_manager.get_dict(ids.contract_state_changes)[
+        ids.tx_info.account_contract_address
+    ]"#
+};
+
+pub fn set_state_entry_to_account_contract_address(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let dict_ptr = get_ptr_from_var_name(vars::ids::CONTRACT_STATE_CHANGES, vm, ids_data, ap_tracking)?;
+    let tx_info_ptr = get_ptr_from_var_name(vars::ids::TX_INFO, vm, ids_data, ap_tracking)?;
+    let account_contract_address =
+        vm.get_integer((tx_info_ptr + TxInfo::account_contract_address_offset())?)?.into_owned();
+
+    get_state_entry(dict_ptr, account_contract_address, vm, exec_scopes, ids_data, ap_tracking)?;
+
+    Ok(())
```

Comment:
Doesn't this need to set the value in `ids.state_entry`?

Reply:
Poor naming on my part, that's what `get_state_entry` does. I changed the name to `set_state_entry`.